### ISSUE TITLE
unit-blacklist ignores values between one and zero without leading-zero

### ISSUE THREAD

**Issue by alunny** [stylelint/stylelint#1773]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug: false-positive for the unit-blacklist rule.

> Which rule, if any, is this issue related to?

`unit-blacklist`

> What CSS is needed to reproduce this issue?

e.g.

```css
.foo {
  font-size: .5rem;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "unit-blacklist": ["rem"]
  }
}
```

> Which version of stylelint are you using?

e.g. `7.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI 

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

Warning should be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

No warnings.

**Comment by jeddy3** [comment]
@alunny Thanks for the report. I can reproduce this.

@stylelint/core This is likely an issue with `getUnitFromNodeValue()` and related to https://github.com/stylelint/stylelint/issues/1771.

**Comment by evilebottnawi** [comment]
@jeddy3 fixed in https://github.com/stylelint/stylelint/pull/1772
